src/api/mod.rs

```diff
@@ -42,81 +43,117 @@ use std::sync::Arc;
   // output size
+  /// Width of the frames in pixels.
   pub width: usize,
+  /// Height of the frames in pixels.
   pub height: usize,
 
   // data format and ancillary color information
+  /// Bit depth.
   pub bit_depth: usize,
+  /// Chroma subsampling.
   pub chroma_sampling: ChromaSampling,
+  /// Chroma sample position.
   pub chroma_sample_position: ChromaSamplePosition,
+  /// Pixel value range.
   pub pixel_range: PixelRange,
+  /// Content color description (primaries, transfer characteristics, matrix).
   pub color_description: Option<ColorDescription>,
+  /// HDR mastering display parameters.
   pub mastering_display: Option<MasteringDisplay>,
+  /// HDR content light parameters.
   pub content_light: Option<ContentLight>,
 
-  /// Still picture mode flag
+  /// Still picture mode flag.
   pub still_picture: bool,
 
   // encoder configuration
+  /// Target CPU feature level.
   pub cpu_feature_level: CpuFeatureLevel,
+  /// Video time base.
   pub time_base: Rational,
   /// The *minimum* interval between two keyframes
   pub min_key_frame_interval: u64,
   /// The *maximum* interval between two keyframes
   pub max_key_frame_interval: u64,
   /// The number of temporal units over which to distribute the reservoir
-  ///  usage.
+  /// usage.
   pub reservoir_frame_delay: Option<i32>,
+  /// Flag to enable low latency mode.
+  ///
+  /// In this mode the frame reordering is disabled.
   pub low_latency: bool,
+  /// The base quantizer to use.
   pub quantizer: usize,
   /// The minimum allowed base quantizer to use in bitrate mode.
   pub min_quantizer: u8,
+  /// The target bitrate for the bitrate mode.
   pub bitrate: i32,
+  /// Metric to tune the quality for.
   pub tune: Tune,
   /// Number of tiles horizontally. Must be a power of two.
-  /// Overridden by the tiles parameter, if present.
+  ///
+  /// Overridden by [`tiles`], if present.
+  ///
+  /// [`tiles`]: #structfield.tiles
   pub tile_cols: usize,
   /// Number of tiles vertically. Must be a power of two.
-  /// Overridden by the tiles parameter, if present.
+  ///
+  /// Overridden by [`tiles`], if present.
+  ///
+  /// [`tiles`]: #structfield.tiles
   pub tile_rows: usize,
-  /// Total number of tiles desired. Encoder will try to optimally split
-  /// to reach this number of tiles, rounded up. Overrides
-  /// tile_cols and tile_rows parameters.
+  /// Total number of tiles desired.
+  ///
+  /// Encoder will try to optimally split to reach this number of tiles,
+  /// rounded up. Overrides [`tile_cols`] and [`tile_rows`].
+  ///
+  /// [`tile_cols`]: #structfield.tile_cols
+  /// [`tile_rows`]: #structfield.tile_rows
   pub tiles: usize,
-  /// Number of frames to read ahead for RDO lookahead computation.
+  /// Number of frames to read ahead for the RDO lookahead computation.
   pub rdo_lookahead_frames: u64,
+  /// Settings which affect the enconding speed vs. quality trade-off.
   pub speed_settings: SpeedSettings,
+  /// Enables printing the PSNR values.
```

Comment:
In terms of the API, it doesn't pint anything - it just attaches the PSNR to the output packet.